### DISCUSSION TITLE
Can an FDMI plugin function as a motr server or only as a motr client?

### DISCUSSION THREAD

**Discussion by johnbent** [Seagate/cortx-motr#1561] [Q&A]
Typical FDMI plugins link against libmotr and can use various client functions. Is it possible for them to also link against internal motr libraries and act as a server?

**Comment by hessio** [comment]
This is a good question I am not actually sure. cc: @huanghua78 can you answer this question?

**Comment by huanghua78** [comment]
> Is it possible for them to also link against internal motr libraries 

If an Motr application is linking against Motr libraries, it can call any functions exported, including functions used by Motr server.
So, this application can do things which are performed by m0d or m0mkfs.
For example, recently, a utility `cas/m0ctgdump` is added to dump a ctg into keys/vals. This works very much like a server process.
But it exits immediately after it is done. Just like m0mkfs.


>and act as a server?

A Motr server or service, will perform some server-like functions, ioservice, mdservice, rmservices.
But any external Motr application, (FDMI application is one of them), does not include any such service defined in the cluster configuration. So other Motr applications and other predefined Motr services will not send their normal Motr requests to it.
Only FDMI records will be posted to it because this is defined in the Motr cluster configuration.
